lib/src/analyzers/lint_analyzer/rules/rules_list/prefer_correct_edge_insets_constructor/visitor.dart

```diff
@@ -0,0 +1,220 @@
+part of 'prefer_correct_edge_insets_constructor_rule.dart';
+
+const _constructorNameFromLTRB = 'fromLTRB';
+const _constructorNameSymmetric = 'symmetric';
+const _constructorNameOnly = 'only';
+
+class _Visitor extends RecursiveAstVisitor<void> {
+  final _expressions = <InstanceCreationExpression, String>{};
+
+  Map<InstanceCreationExpression, String> get expressions => _expressions;
+
+  @override
+  void visitInstanceCreationExpression(InstanceCreationExpression expression) {
+    super.visitInstanceCreationExpression(expression);
+
+    final constructorName = expression.constructorName.name?.name;
+    if (constructorName == _constructorNameFromLTRB) {
+      _validateLTRB(expression);
+    } else if (constructorName == _constructorNameSymmetric) {
+      _validateSymmetric(expression);
+    } else if (constructorName == _constructorNameOnly) {
+      _validateOnly(expression);
+    }
+  }
+
+  void _validateLTRB(InstanceCreationExpression expression) {
+    if (!expression.argumentList.arguments.every(
+      (element) => element is IntegerLiteral || element is DoubleLiteral,
+    )) {
+      return;
+    }
+
+    final argumentsList =
+        expression.argumentList.arguments.map((e) => e.toString());
+
+    // EdgeInsets.fromLTRB(12,12,12,12) -> EdgeInsets.all(12)
+    if (argumentsList.every((element) => element == argumentsList.first)) {
+      _expressions[expression] = 'EdgeInsets.all(${argumentsList.first})';
+
+      return;
+    }
+
+    // EdgeInsets.fromLTRB(3,2,3,2) -> EdgeInsets.symmetric(horizontal: 3, vertical: 2)
+    if (argumentsList.length == 4 &&
+        argumentsList.first == argumentsList.elementAt(2) &&
+        argumentsList.elementAt(1) == argumentsList.elementAt(3)) {
+      final params = <String>[];
+      if (num.tryParse(argumentsList.first) != 0) {
+        params.add(
+          'horizontal: ${argumentsList.first}',
+        );
+      }
+      if (num.tryParse(argumentsList.elementAt(1)) != 0) {
+        params.add(
+          'vertical: ${argumentsList.elementAt(1)}',
+        );
+      }
+      _expressions[expression] = 'EdgeInsets.symmetric(${params.join(', ')})';
+
+      return;
+    }
+
+    // EdgeInsets.fromLTRB(3,0,0,2) -> EdgeInsets.only(left: 3, bottom: 2)
+    if (argumentsList.contains('0') || argumentsList.contains('0.0')) {
```

Comment:
So, what will happen for `EdgeInsets.fromLTRB(0,0,0,0)`? Or it's prohibited to use?

Reply:
This case is already being processed by a linter from a box

Reply:
So, why not to suggest `EdgeInsets.zero` then?